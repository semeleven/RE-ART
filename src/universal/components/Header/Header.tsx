import React, { Component, Fragment } from 'react';
import styled from '../../lib/styled';

import { Button, Icon, SideMenu } from '../../components';

interface State {
	showMenu: boolean
}

const StyledHeader = styled.header`
	display: flex;
	color: ${props => props.theme.white};
	flex-direction: row;
	justify-content: space-between;
	align-items: center;
	background-color: ${props => props.theme.darkGray};
	padding-left: 24px;
	padding-right: 24px;
	height: 82px;
	margin-bottom: 50px;
`;

const Wrapper = styled.div`
	display: flex;
	flex-direction: row;
	width: 50%;
	justify-content: flex-end;
	align-items: center;
`;

class Header extends Component<any, State> {
	state = {
		showMenu: false,
	};

	handleMenu = () => this.setState(state => ({ showMenu: !state.showMenu }));

	render() {
		const { showMenu } = this.state;

		return (
			<Fragment>
				<StyledHeader>
					<Icon icon="hamburger" onClick={this.handleMenu} />
					<Wrapper>
						<Icon small={false} icon="logo" />
					</Wrapper>
					<Wrapper>
					<Button invertedPurple onClick={() => console.log('LOGIN')} spaced>
						LOGIN
					</Button>
					<Button purple onClick={() => console.log('SIGN UP')} spaced>
						SIGN UP
					</Button>
						<Icon icon="cart" />
					</Wrapper>
				</StyledHeader>
				<SideMenu showMenu={showMenu}>
					<h1>HELLO WORLD</h1>
				</SideMenu>
			</Fragment>
		);
	}
}

export default Header;
