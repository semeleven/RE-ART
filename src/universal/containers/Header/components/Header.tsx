import React, { Fragment } from 'react';
import styled, { css } from '@Styled';

import { Button, Icon, TopMenu, Link } from '@Components';
import { colors, media } from '@Styled/theme';

interface Props {
	toggleModal: () => void;
	toggleMenu: () => void;
	showMenu: boolean;
	token?: string; // user's token
}

const StyledHeader = styled.header`
	display: flex;
	justify-content: space-between;
	align-items: center;
	background-color: ${colors.darkGray};
	padding-left: 24px;
	padding-right: 24px;
	height: 82px;
	margin-bottom: 50px;
	${media.mobile`
		padding-left: 12px;
		padding-right: 12px;
	`};
`;

interface FlexWrapperProps {
	flex: 'start' | 'end';
}

const FlexWrapper = styled.div`
	${(props: FlexWrapperProps) =>
		css`
			display: flex;
			width: 100%;
			flex: 1;
			align-items: center;
			justify-content: ${props.flex === 'start' ? 'flex-start' : 'flex-end'};
		`};
`;

const LogoWrapper = styled.div`
	display: flex;
	width: 100%;
	flex: 2;
	align-items: center;
	justify-content: center;
`;

const HideOnMobile = styled.div`
	${media.mobile`
		display: none;
	`};
`;

const ShowOnMobileOnly = styled.div`
	display: none;
	${media.mobile`
		display: flex;
		color: white;
		padding-right: 20px;
	`};
`;

const PersonalWrapper = styled.div`
	display: flex;
	margin-right: 60px;
	${media.tablet`
		margin-right: 30px;
	`};
`;

const IconWrapper = styled.div`
	margin-left: 30px;
`;

export const Authorization = ({ toggleModal }) => (
	<Fragment>
		<ShowOnMobileOnly>
			{/* render login link instead of button on mobile devices */}
			<Link white onClick={toggleModal} size="S">
				LOGIN
			</Link>
		</ShowOnMobileOnly>
		<HideOnMobile>
			<Button purple onClick={toggleModal} spaced>
				LOGIN
			</Button>
		</HideOnMobile>
	</Fragment>
);

export const Personal = () => (
	<HideOnMobile>
		<PersonalWrapper>
			<IconWrapper>
				<Icon icon="Chat" />
			</IconWrapper>
			<IconWrapper>
				<Icon icon="Notifications" />
			</IconWrapper>
			<IconWrapper>
				<Icon icon="Account" />
			</IconWrapper>
		</PersonalWrapper>
	</HideOnMobile>
);

const Header: React.SFC<Props> = ({
	showMenu,
	toggleModal,
	toggleMenu,
	token,
}) => (
	<Fragment>
		<StyledHeader>
			<FlexWrapper flex="start">
				<Icon icon="Hamburger" onClick={toggleMenu} />
			</FlexWrapper>
			<LogoWrapper>
				<Icon small={false} icon="Logo" />
			</LogoWrapper>
			<FlexWrapper flex="end">
				{token ? <Personal /> : <Authorization toggleModal={toggleModal} />}
				<Icon icon="Cart" />
			</FlexWrapper>
		</StyledHeader>
		<TopMenu showMenu={showMenu}>
			<h1>HELLO WORLD</h1>
		</TopMenu>
	</Fragment>
);

export default Header;
