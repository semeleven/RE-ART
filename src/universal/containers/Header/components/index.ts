import Header from './Header';
import TopMenu from './TopMenu';

export { Header, TopMenu };
