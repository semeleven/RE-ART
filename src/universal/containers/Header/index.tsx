import React, { PureComponent } from 'react';
// import { Dispatch, bindActionCreators } from 'redux';
import { connect } from 'react-redux';

import {
	getUserAndLayout,
	getUserAndLayoutType,
} from '../../lib/redux/reselect';

import { mapDispatchToProps } from '../../lib/redux/helpers';

import {
	ToggleModal,
	ToggleMenu,
} from '../../lib/redux/reducers/Layout/LayoutActions';

import Header from './components/Header';

interface Actions {
	toggleModal: () => void;
	toggleMenu: () => void;
}

class HeaderContainer extends PureComponent<getUserAndLayoutType & Actions> {
	render() {
		const {
			toggleModal,
			toggleMenu,
			layout: { showMenu },
		} = this.props;

		return (
			<Header
				showMenu={showMenu}
				toggleModal={toggleModal}
				toggleMenu={toggleMenu}
			/>
		);
	}
}

const actionCreators = {
	toggleModal: ToggleModal,
	toggleMenu: ToggleMenu,
};

export default connect(
	getUserAndLayout,
	mapDispatchToProps(actionCreators)
)(HeaderContainer);
