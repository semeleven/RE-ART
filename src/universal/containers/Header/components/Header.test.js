import React from 'react';
import { shallow } from 'enzyme';
import 'jest-styled-components';

import Header, { Authorization, Personal } from './Header';

const noop = () => {};

it('renders Authorization block', () => {
	const wrapper = shallow(<Authorization />);
	expect(wrapper).toMatchSnapshot();
});

it('renders Personal block', () => {
	const wrapper = shallow(<Personal />);
	expect(wrapper).toMatchSnapshot();
});

it('renders Header for an unauthorized user', () => {
	const wrapper = shallow(
		<Header showMenu={false} toggleModal={noop} toggleMenu={noop} />
	);
	expect(wrapper).toMatchSnapshot();
});

it('renders Header for an authorized user', () => {
	const wrapper = shallow(
		<Header
			showMenu={false}
			toggleModal={noop}
			toggleMenu={noop}
			token="token"
		/>
	);
	expect(wrapper).toMatchSnapshot();
});

it('renders Top Menu', () => {
	const wrapper = shallow(
		<Header showMenu toggleModal={noop} toggleMenu={noop} token="token" />
	);
	expect(wrapper).toMatchSnapshot();
});
