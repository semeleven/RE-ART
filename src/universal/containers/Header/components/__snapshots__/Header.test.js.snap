// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Authorization block 1`] = `
<React.Fragment>
  <styled.div>
    <Styled(LinkComponent)
      size="S"
      white={true}
    >
      LOGIN
    </Styled(LinkComponent)>
  </styled.div>
  <styled.div>
    <Styled(Button)
      purple={true}
      spaced={true}
    >
      LOGIN
    </Styled(Button)>
  </styled.div>
</React.Fragment>
`;

exports[`renders Header for an authorized user 1`] = `
<React.Fragment>
  <styled.header>
    <styled.div
      flex="start"
    >
      <Icon
        icon="Hamburger"
        onClick={[Function]}
        pointer={true}
        small={true}
      />
    </styled.div>
    <styled.div>
      <Icon
        icon="Logo"
        onClick={[Function]}
        pointer={true}
        small={false}
      />
    </styled.div>
    <styled.div
      flex="end"
    >
      <Component />
      <Icon
        icon="Cart"
        onClick={[Function]}
        pointer={true}
        small={true}
      />
    </styled.div>
  </styled.header>
  <TopMenu
    showMenu={false}
  >
    <h1>
      HELLO WORLD
    </h1>
  </TopMenu>
</React.Fragment>
`;

exports[`renders Header for an unauthorized user 1`] = `
<React.Fragment>
  <styled.header>
    <styled.div
      flex="start"
    >
      <Icon
        icon="Hamburger"
        onClick={[Function]}
        pointer={true}
        small={true}
      />
    </styled.div>
    <styled.div>
      <Icon
        icon="Logo"
        onClick={[Function]}
        pointer={true}
        small={false}
      />
    </styled.div>
    <styled.div
      flex="end"
    >
      <Component
        toggleModal={[Function]}
      />
      <Icon
        icon="Cart"
        onClick={[Function]}
        pointer={true}
        small={true}
      />
    </styled.div>
  </styled.header>
  <TopMenu
    showMenu={false}
  >
    <h1>
      HELLO WORLD
    </h1>
  </TopMenu>
</React.Fragment>
`;

exports[`renders Personal block 1`] = `
<styled.div>
  <styled.div>
    <styled.div>
      <Icon
        icon="Chat"
        onClick={[Function]}
        pointer={true}
        small={true}
      />
    </styled.div>
    <styled.div>
      <Icon
        icon="Notifications"
        onClick={[Function]}
        pointer={true}
        small={true}
      />
    </styled.div>
    <styled.div>
      <Icon
        icon="Account"
        onClick={[Function]}
        pointer={true}
        small={true}
      />
    </styled.div>
  </styled.div>
</styled.div>
`;

exports[`renders Top Menu 1`] = `
<React.Fragment>
  <styled.header>
    <styled.div
      flex="start"
    >
      <Icon
        icon="Hamburger"
        onClick={[Function]}
        pointer={true}
        small={true}
      />
    </styled.div>
    <styled.div>
      <Icon
        icon="Logo"
        onClick={[Function]}
        pointer={true}
        small={false}
      />
    </styled.div>
    <styled.div
      flex="end"
    >
      <Component />
      <Icon
        icon="Cart"
        onClick={[Function]}
        pointer={true}
        small={true}
      />
    </styled.div>
  </styled.header>
  <TopMenu
    showMenu={true}
  >
    <h1>
      HELLO WORLD
    </h1>
  </TopMenu>
</React.Fragment>
`;
